marlowe-playground-client/src/Halogen/ActusBlockly.purs

```diff
@@ -152,7 +154,16 @@ handleAction (GetTerms flavour) = do
     Left e -> assign _errorMessage $ Just e
     Right contract -> do
       assign _errorMessage Nothing
-      raise $ CurrentTerms flavour $ contract
+      case parseActusJsonCode contract of
+        Left e -> assign _errorMessage $ Just e
+        Right ctRaw -> do
+          let ct = (unwrap ctRaw)
+          case flavour of
+            F -> if (isJust ct.ct_RRCL) 
+                    then assign _errorMessage $ Just "Rate resets are not allowed in static contracts"
+                    else raise $ CurrentTerms flavour $ contract
+            _ -> raise $ CurrentTerms flavour $ contract
```

Comment:
It would be nice to use ExcepT for everything here, for example, would that be possible? Just a thought